src/Jackalope/Transport/DoctrineDBAL/XmlPropsRemover/XmlPropsRemover.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace Jackalope\Transport\DoctrineDBAL\XmlPropsRemover;
+
+/**
+ * @internal
+ */
+class XmlPropsRemover
+{
+    private string $xml;
+
+    /**
+     * @var string[]
+     */
+    private array $propertyNames;
+
+    private bool $skipCurrentTag = false;
+
+    private string $newXml = '';
+
+    private string $newStartTag = '';
+
+    /**
+     * @var string[]
+     */
+    private array $weakReferences = [];
+
+    /**
+     * @var string[]
+     */
+    private array $binaries = [];
+
+    /**
+     * @var string[]
+     */
+    private array $references = [];
+
+    public function __construct(string $xml, array $propertyNames)
+    {
+        $this->xml = $xml;
+        $this->propertyNames = $propertyNames;
+    }
+
+    /**
+     * @example [$xml, $references, $binaries] = $xmlPropsRemover->removeProps($xml, $propertiesToDelete);
+     *
+     * @return array{
+     *     0: string,
+     *     1: array{
+     *         reference: string[],
+     *         weakreference: string[],
+     *     },
+     *     2: string[]
+     * } An array with the new xml (0) and the references (1) which requires to be removed
+     */
+    public function removeProps(): array
+    {
+        $this->newXml = '<?xml version="1.0" encoding="UTF-8"?>'.PHP_EOL;
+        $this->references = [];
+        $this->weakReferences = [];
+        $this->binaries = [];
+        $this->newStartTag = '';
+        $this->skipCurrentTag = false;
+
+        $parser = \xml_parser_create();
+
+        \xml_set_element_handler(
+            $parser,
+            [$this, 'startHandler'],
+            [$this, 'endHandler']
+        );
+
+        \xml_set_character_data_handler($parser, [$this, 'dataHandler']);
+
+        \xml_parse($parser, $this->xml, true);
+        \xml_parser_free($parser);
+        // avoid memory leaks and unset the parser see: https://www.php.net/manual/de/function.xml-parser-free.php
+        unset($parser);
+
+        return [
+            $this->newXml.PHP_EOL,
+            [
+                'reference' => $this->references,
+                'weakreference' => $this->weakReferences,
+            ],
+            $this->binaries,
+        ];
+    }
+
+    /**
+     * @param mixed[] $attrs
+     */
+    private function startHandler(\XMLParser $parser, string $name, array $attrs): void
+    {
+        if ($this->skipCurrentTag) {
+            return;
+        }
+
+        if ('SV:PROPERTY' === $name) {
+            $svName = $attrs['SV:NAME'];
+
+            if (\in_array((string)$svName, $this->propertyNames, true)) {
+                $this->skipCurrentTag = true;
+                $svType = strtolower($attrs['SV:TYPE']);
```

Comment:
with the binary removal test, the type was `Binary` and not `binary`. the node type definition have capitalized names for all types, but i guess it already worked for references (or is that not covered by the tests?)

not sure what exactly is going on. if you have an idea @alexander-schranz we could clean it up, but otherwise it should work like this.

Reply:
the tests seems to use lower case part. But I copied that from another tests so not sure if it was already there false.